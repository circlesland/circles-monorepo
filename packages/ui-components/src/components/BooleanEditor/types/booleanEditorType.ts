export enum BooleanEditorType {
    Checkbox,
    Toggle
}