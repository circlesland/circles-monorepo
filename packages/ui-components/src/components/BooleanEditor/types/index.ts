export * from './booleanEditorType';
export * from './customBooleanEditorTheme';