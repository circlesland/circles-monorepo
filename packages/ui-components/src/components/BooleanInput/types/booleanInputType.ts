export enum BooleanInputType {
    Checkbox,
    Toggle
}