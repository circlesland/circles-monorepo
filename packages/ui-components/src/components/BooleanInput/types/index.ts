export * from './booleanInputType';
export * from './customBooleanInputTheme';