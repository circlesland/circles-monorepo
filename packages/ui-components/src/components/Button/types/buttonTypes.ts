export enum ButtonTypes {
    Default,
    IconText,
    CircularWithIcon
}