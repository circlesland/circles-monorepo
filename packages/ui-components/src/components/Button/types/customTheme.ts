export interface CustomButtonTheme {
    width?: number;
    height?: number;
}