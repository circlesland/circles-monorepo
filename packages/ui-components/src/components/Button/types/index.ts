export * from './buttonTypes';
export * from './customTheme';