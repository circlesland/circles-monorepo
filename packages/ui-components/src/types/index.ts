export * from './view';
export * from './supported';