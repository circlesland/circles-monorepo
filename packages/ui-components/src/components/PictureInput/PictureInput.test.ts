/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/svelte';

import { ViewType } from '../../types';
import PictureInputSvelte from './PictureInput.svelte';

import type { View, PictureInput } from '../../types';
describe('Picture Input', () => {
  const pictureInput: View & PictureInput = {
    id: 'picture-input',
    testId: 'picture-input-test-id',
    type: ViewType.PICTURE_INPUT,
    args: {},
  };

  it('renders correctly', () => {
    const tree = render(PictureInputSvelte, { view: pictureInput });
    expect(tree).toMatchSnapshot();
  });
});
