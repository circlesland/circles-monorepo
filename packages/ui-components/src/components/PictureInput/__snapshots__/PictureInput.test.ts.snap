// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Input renders correctly 1`] = `
{
  "component": PictureInput {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        {
          "args": {},
          "id": "picture-input",
          "testId": "picture-input-test-id",
          "type": "PictureInput",
        },
        undefined,
        <input
          accept=".jpg, .jpeg, .png"
          data-testid="picture-input-test-id-file-input"
          type="file"
        />,
        undefined,
        undefined,
        "picture-input-test-id",
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "view": 0,
      },
      "root": <div>
        <div
          data-testid="picture-input-test-id"
        >
          <h2>
            Upload a picture for cropping
          </h2>
           
          <input
            accept=".jpg, .jpeg, .png"
            data-testid="picture-input-test-id-file-input"
            type="file"
          />
          
        </div>
        
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        data-testid="picture-input-test-id"
      >
        <h2>
          Upload a picture for cropping
        </h2>
         
        <input
          accept=".jpg, .jpeg, .png"
          data-testid="picture-input-test-id-file-input"
          type="file"
        />
        
      </div>
      
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
